app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/roidmi/RoidmiSupport.java

```diff
@@ -0,0 +1,139 @@
+/*  Copyright (C) 2018 José Rebelo
+
+    This file is part of Gadgetbridge.
+
+    Gadgetbridge is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Gadgetbridge is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+package nodomain.freeyourgadget.gadgetbridge.service.devices.roidmi;
+
+import android.net.Uri;
+import android.os.Handler;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+import nodomain.freeyourgadget.gadgetbridge.devices.roidmi.RoidmiConst;
+import nodomain.freeyourgadget.gadgetbridge.model.Alarm;
+import nodomain.freeyourgadget.gadgetbridge.model.DeviceType;
+import nodomain.freeyourgadget.gadgetbridge.service.serial.AbstractSerialDeviceSupport;
+import nodomain.freeyourgadget.gadgetbridge.service.serial.GBDeviceIoThread;
+import nodomain.freeyourgadget.gadgetbridge.service.serial.GBDeviceProtocol;
+
+public class RoidmiSupport extends AbstractSerialDeviceSupport {
+    private static final Logger LOG = LoggerFactory.getLogger(RoidmiSupport.class);
+
+    @Override
+    public boolean connect() {
+        getDeviceIOThread().start();
+
+        // FIXME not like this
```

Comment:
@cpfeiffer before merging I still need to find a more elegant solution for this, in order to handle an eventual timeout, and also periodically poll the battery voltage.

Do you know if there is any mechanism already built into Gadgetbridge that would allow me to do that?

Reply:
Other devices start sending battery infos automatically when it gets low.

However, we might also get some generic infrastructure for this, where the UI periodically requests a device update for all connected devices. Either when the display is turned on and Gadgetbridge is in the foreground, or really even periodically when the display is on all the time.

The interval for such update requests might be a device specific value specified by the DeviceCoordinator.

What do you mean exactly with a timeout? Automatic reconnection when a timeout occurs?

Reply:
Here the values are only sent on request. There's some interest in polling the battery voltage, for eventual warnings if it gets too low / too high (possibly with a customizable threshold).

I am not sure if I am comfortable enough with Gadgetbridge's codebase to implement something like this (especially in a battery-efficient / doze-respecting way) but I can give it a shot.

By timeout I mean requesting the FM Frequency / LED Color again if no answer is received. There should be no need to periodically poll them like the battery voltage, since they will not change, but sometimes the device will not answer the requests, especially if they're made right after the initial connection (hence the Handler with the 3~5 second delay, it almost never fails).

Reply:
I've opened #1189 for the device info updating. I think there shouldn't be any issues regarding battery-friendliness, since we would only do this when the display is active or activated (unless you have other requirements).